### ISSUE TITLE
k8s Gateway API support

### ISSUE THREAD

**Issue by gemelen** [hetznercloud/hcloud-cloud-controller-manager#814]
### TL;DR

Add support for the [Gateway API](https://gateway-api.sigs.k8s.io) in the LoadBalancer managing part of the CCM, so its `Gateway` component would be provided with the LoadBalancer information on par with a `LoadBalancer` service type.

This would handle the cases when a k8s cluster is deployed and configured (exclusively or preferably) with the Gateway API approach of exposing services. That covers setups managed by latest versions of Cilium, Traefik, Nginx and others, see [list of implementations](https://gateway-api.sigs.k8s.io/implementations/) and compatibility tables for [v1.2](https://gateway-api.sigs.k8s.io/implementations/v1.2/), [v1.1](https://gateway-api.sigs.k8s.io/implementations/v1.1/).

One related [issue](https://github.com/hetznercloud/hcloud-cloud-controller-manager/issues/756) has already been raised here.

### Expected behavior

In a cluster setup configured in a Gateway API fashion, a [`Gateway`](https://gateway-api.sigs.k8s.io/guides/simple-gateway/) should be able to obtain an external address(es) from an (external) load balancer, created by or provided via Hetzner CCM.

**Comment by zfrgt** [comment]
that would be nice to have

**Comment by am-i-helpful** [comment]
Hi!
I was able to provision an external LB in Hetzner-Cloud using the Cilium Gateway-API following the public documentation of hccm I found here (https://github.com/vitobotta/hetzner-k3s/blob/main/docs/Load_balancers.md#load-balancers), and a bit of manual work annotating the service derived as part of 'Gateway' resource which I think is a bug at Cilium's end: https://github.com/cilium/cilium/issues/36792.

In my opinion, the limitation is at the Cilium Gateway-API controller that they are not copying the metadata 'annotation' from the 'Gateway' resource to the provisioned service of type 'LoadBalancer', which would otherwise result in hccm to provision an external LoadBalancer automatically.

**Comment by kishoreinvits** [comment]
@gemelen , @am-i-helpful 

If you are okay with experimental CRDs, annotations work with cilium as mentioned in [this cilium issue comment](https://github.com/cilium/cilium/issues/25357#issuecomment-2259214557).

Note that [cilium docs 1.16](https://docs.cilium.io/en/v1.16/network/servicemesh/gateway-api/gateway-api/#gs-gateway-api) (latest at the time of writing) asks to use v1.1.0 Gateway API CRDs, so I don't recommend using v1.2.0 experimntal as mentioned in that comment until cilium specifies it.

**Comment by lukasmetzner** [comment]
Hey,

the hcloud-cloud-controller-manager based on the general concept of CCMs is not responsible for the Gateway API. This would require the implementation of a new Gateway Controller, which is currently not planned.

Best Regards,
Lukas

**Comment by lukasmetzner** [comment]
@deubert-it When creating a new tutorial for a Gateway API example, I will also try to put a small explanation into `docs/explanation/...`. This way users, have a more structured way, of finding this information :D
